tc/src/main/java/com/epam/tc/security/AuthenticatedUser.java

```diff
@@ -1,14 +1,23 @@
 package com.epam.tc.security;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.stereotype.Service;
 
 @Service
 public class AuthenticatedUser {
 
-    public String getUserName() {
-        UserDetails ud = (UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
-        return ud.getUsername();
+    private static final Logger LOG = LoggerFactory.getLogger(AuthenticatedUser.class);
+
+    public String getUserEmail() {
+        try {
+            UserDetails ud = (UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+            return ud.getUsername();
+        } catch (ClassCastException ex) {
```

Comment:
Catching of ClassCastExceptoin looks a little bit strange. Can you do some checks before casting?